aws_advanced_python_wrapper/host_monitoring_v2_plugin.py

```diff
@@ -0,0 +1,506 @@
+                    self._monitoring_host_info = self._plugin_service.identify_connection(current_connection)
+                    if self._monitoring_host_info is None:
+                        raise AwsWrapperError(
+                            Messages.get_formatted(
+                                "HostMonitoringV2Plugin.UnableToIdentifyConnection",
+                                current_host_info.host,
+                                self._plugin_service.host_list_provider))
+                    self._plugin_service.fill_aliases(current_connection, self._monitoring_host_info)
+            except Exception as e:
+                message = "HostMonitoringV2Plugin.ErrorIdentifyingConnection"
+                logger.debug(message, e)
+                raise AwsWrapperError(Messages.get_formatted(message, e)) from e
+        return self._monitoring_host_info
+
+    def release_resources(self):
+        if self._monitor_service is not None:
+            self._monitor_service.release_resources()
+
+        self._monitor_service = None
+
+
+class HostMonitoringV2PluginFactory(PluginFactory):
+
+    def get_instance(self, plugin_service: PluginService, props: Properties) -> Plugin:
+        return HostMonitoringV2Plugin(plugin_service, props)
+
+
+class MonitoringContext:
+    """
+    Monitoring context for each connection.
+    This contains each connection's criteria for whether a server should be considered unhealthy.
+    The context is shared between the main thread and the monitor thread.
+    """
+
+    def __init__(self, connection: Connection):
+        self._connection_to_abort: AtomicReference = AtomicReference(weakref.ref(connection))
+        self._node_unhealthy: AtomicBoolean = AtomicBoolean(False)
+
+    def set_node_unhealthy(self) -> None:
+        self._node_unhealthy.set(True)
+
+    def should_abort(self):
+        connection_weak_ref = self._connection_to_abort.get()
+        return self._node_unhealthy.get() and connection_weak_ref is not None and connection_weak_ref() is not None
+
+    def set_inactive(self) -> None:
+        self._connection_to_abort.set(None)
+
+    def get_connection(self) -> Connection | None:
+        connection_weak_ref = self._connection_to_abort.get()
+        if connection_weak_ref is not None:
+            return connection_weak_ref()
+        else:
+            return None
+
+    def is_active(self) -> bool:
+        connection_weak_ref = self._connection_to_abort.get()
+        return connection_weak_ref is not None and connection_weak_ref() is not None
+
+
+class HostMonitorV2:
+    """
+        This class uses a background thread to monitor a particular server with one or more active :py:class:`Connection`
+        objects. It performs periodic health checks and aborts connections when the server becomes unhealthy.
+        """
+    _THREAD_SLEEP_NANO = 100_000_000
+    _MONITORING_PROPERTY_PREFIX = "monitoring-"
+    _QUERY = "SELECT 1"
+
```

Comment:
Please check the usage of these. Should use them as static variables instead of `self._...`